### PR TITLE
Types and Data modeling

### LINKED ISSUE #170: 'Designing Data Models' Section
The new section 'designing data models' is important enough that I think it needs a top-level distinction.

Introduction

Installation
-> Swift
-> JS
-> Rust
..etc

Concepts
-> Documents
-> Update
-> Upset
-> Querying
-> Remove
-> Evict
-> Syncing Data
-> Big Peer

Designing Data Models
-> Counters
-> Arrays
-> Attachments and Binary data
-> Going offline
-> Versioning
-> Continual Eviction

Guides 
-> Kafka
-> Authentication
-> iOS Quick Tips
-> HTTP API
-> Network Configuration
-> Security 

Quickstart (formally Tutorials->Tasks App)
-> Swift
-> Android
-> Jetpack compose
-> etc...

How it works
-> CRDTs
-> Mesh Network
-> Certificate Based Security
-> Supported Network Transports
-> Big Peer

FAQ

API Reference
-> Swift
-> JS
-> Rust
..etc

### PR DESCRIPTION
**Pull request by okdistribute**
We have a lot of conversations with users about how our types work. It isn't always obvious. To give a better reference point, we call this out specifically in a Types sidebar item.

Closes #170 #11

### CODE REVIEW

File: docs/types/array.mdx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import SnippetGroup from '@site/src/components/SnippetGroup';
+
+Arrays are an ordered list of values. Arrays can contain all primitive
+values as well as nested collection types like other Arrays or Maps
+
+* Push - upserts a value on to the end of an array at the document's key
```
Comment (russelldb) [inline docs/types/array.mdx:13]: Are these really the only supported operations? What about update-at-index? Insert-at-index?

Comment (okdistribute) [reply docs/types/array.mdx:13]: I'm just copying this from the current website, but I'll take a look. Users definitely want update-at and insert-at

**Comment by arthurprs** [comment]
I'm late to the PR but I just read it and it looks great :love_letter:

**Comment by russelldb** [comment]
I struggled to get my review done in time for the merge. Is this now live on ditto.live? I also struggled to see the files out of context.

**Comment by okdistribute** [comment]
Yes it's live now on https://docs.ditto.live

**Pull request merged**